openapi/admin/userGroup.yaml

```diff
@@ -0,0 +1,94 @@
+get:
+  tags:
+  - "UserGroup"
+  operationId: "getAllUserGroups"
+  description: "Finds all user groups"
+  responses:
+    "200":
+      description: "User groups successfully fetched"
+      content:
+        application/json:
+          schema:
+            type: "array"
+            items:
+              $ref: "../schemas/admin/UserGroupResponse.yaml"
+    "401":
+      $ref: "../responses/UnauthorizedError.yaml"
+    "403":
+      $ref: "../responses/ForbiddenError.yaml"
+post:
+  tags:
+  - "UserGroup"
+  operationId: "createUserGroup"
+  description: "Create a new user group"
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: "../schemas/admin/UserGroupCreateRequest.yaml"
+  responses:
+    "200":
+      description: "User group was successfully created"
+      content:
+        application/json:
+          schema:
+            $ref: "../schemas/DefaultCreateResponse.yaml"
+    "401":
+      $ref: "../responses/UnauthorizedError.yaml"
+    "403":
+      $ref: "../responses/ForbiddenError.yaml"
+put:
+  tags:
+  - "UserGroup"
+  operationId: "updateUserGroup"
+  description: "Update an existing user group"
+  parameters:
+  - name: "userGroupId"
+    description: "ID of the user group to update"
+    required: true
+    in: query
+    schema:
+      type: "integer"
+      format: "int64"
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: "../schemas/admin/UserGroupCreateRequest.yaml"
```

Comment:
Could we name the `UserGroupCreateRequest` something more operation-agnostic? Maybe just `UserGroupData`. It feels a bit off to use something named `create` for an update request. We could also introduce an `UserGroupUpdateRequest` which contains `userGroupId`, but that might be a bit redundant.